app/models/miq_report_result.rb

```diff
@@ -339,12 +339,17 @@ def self.delete_by_userid(userids)
   def self.find_all_by_users_group(userid)
     user = User.find_by_userid(userid)
     return [] if user.nil? || user.miq_group.nil?
-    miq_report_ids = MiqReport.where("miq_group_id = ?", user.miq_group.id).pluck("id")
+    miq_report_ids = MiqReport.where(:miq_group_id => user.miq_group.id).pluck("id")
 
-    MiqReportResult.all(
-      :conditions => ["miq_report_id IN (?) AND report_source != ?", miq_report_ids, "Generated by user"],
-      :select     => "miq_report_id, name",
-      :group      => "miq_report_id, name"
-    )
+    MiqReportResult.not_user_generated("miq_report_id, name").where(:miq_report_id => miq_report_ids)
+  end
+
+  def self.not_user_generated(group_by = nil, select = group_by)
+    rslts = where.not(:report_source => "Generated by user")
+    if group_by
+      rslts.select(select).group(group_by)
```

Comment:
While there is similarity across the callers, I don't see how the `select`/`group` is in any way related to the `not_user_generated` scope.

`not_user_generated('xx')` vs `not_user_generated.select('xx').distinct` -- for the extra characters, the clarity seems worthwhile. If it's still useful to extract the select+distinct / select+group, I think it'd be better in a separate method.